server/dataone_register.py

```diff
@@ -40,14 +37,33 @@ def unesc(value):
     return urllib.parse.unquote_plus(value)
 
 
-def query(q, fields=["identifier"], rows=1000, start=0):
-    """Query a DataONE Solr index."""
+def query(q,
```

Comment:
Now that we're using `d1_python`, we could also use `d1_python`'s Solr functionality which might be most robust: https://github.com/DataONEorg/d1_python/blob/f6a7d881f4705bdef46c076f762b5d8f400cfc5f/lib_client/src/d1_client/solr_client.py

I probably wouldn't actually do this just to save time in doing it but it's an idea.

Reply:
Nice find; a refactor wouldn't be too much of a hassle because the change would be in 
`def query`, which is already called in the right places.